src/Grpc.Net.Client/Balancer/Internal/ConnectionManager.cs

```diff
@@ -468,7 +468,7 @@ void OnCancellation(object? s)
                     StateWatcher watcher = (StateWatcher)s!;
                     if (_stateWatchers.Remove(watcher))
                     {
-                        watcher.Tcs.SetCanceled();
+                        watcher.Tcs.SetCanceled(cancellationToken);
```

Comment:
This introduces a closure. Need to refactor this method to take cancellation token as an argument. I think that is available in newer .nets